### PR TITLE
Add diff syntax highlighting

### PR DESCRIPTION
**Pull request by RubenSandwich**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-rn-fixadd-diff-syntax-highlighting-hashicorp.vercel.app/) 🔎
- [Asana task](https://app.asana.com/0/1205890531436279/1207516355156788/f) 🎟️

## 🗒️ What

Adding syntax highlighting to diff code blocks.

## 📸 Design Screenshots

Screenshot 2024-06-07 at 1 03 40 PM

## 🧪 Testing

- [x] Make sure the the diff codeblock on the page [terraform/tutorials/configuration-language/count#refactor-the-ec2-configuration](https://dev-portal-git-rn-fixadd-diff-syntax-highlighting-hashicorp.vercel.app/terraform/tutorials/configuration-language/count#refactor-the-ec2-configuration) is highlighted

### CODE REVIEW

File: src/lib/rehype-code-plugins.ts [hunk 1]
```diff
@@ -41,6 +43,16 @@ export const rehypeCodePlugins: Pluggable[] = [
 							this.addClassToHast(node, 'empty-line')
 						}
 					},
+					span(node, _line: number, col: number) {
+						// prevent user-select on the start +/- in diff lines
+						if (this.options.lang === 'diff' && col === 0) {
+							const firstChild = node.children[0] as any
```
Comment (RubenSandwich) [inline src/lib/rehype-code-plugins.ts:49]: Not happy with this, but TS complained when I casted to Element. 😢

Comment (EnMod) [reply src/lib/rehype-code-plugins.ts:49]: 👎 at TS for this! I was just about to suggest Element 😅

File: src/lib/syntax-highlighting/theme.ts [hunk 2]
```diff
@@ -135,5 +135,23 @@ export const theme: ThemeRegistrationResolved = {
...
+		},
+		{
+			scope: [
+				'punctuation.definition.deleted.diff',
+				'punctuation.definition.inserted.diff',
+			],
+			settings: { foreground: 'var(--hds-code-block-color-token)' },
+		},
+		// TODO: We should map our code block colors to the correct scope:
+		// https://github.com/hashicorp/design-system/blob/2c3c9a6a317255c3c8e22d72aa327eeaa7ccc7f3/packages/components/src/styles/components/code-block/theme.scss#L8
```
Comment (EnMod) [inline src/lib/syntax-highlighting/theme.ts:155]: Great comments all around ✨

**Review by EnMod — APPROVED** [review]
LGTM! Especially those comments 💯

**Comment by RubenSandwich** [comment]
Will wait till Monday to push as we have run into a bunch of issues recently with our codeblocks.

**Pull request merged**